### PR TITLE
Implement per-user package directory feature

### PR DESCRIPTION
**Pull request by kyujin-cho**
This PR resolves issue lablup/backend.ai#99.

### CODE REVIEW

**Comment by achimnol** [comment]
OP#106

**Comment by achimnol** [comment]
Ticket: https://op.lablup.ai:7443/projects/backend-ai-server-and-sdk/work_packages/106/activity

**Comment by achimnol** [comment]
mic test

**Comment by achimnol** [comment]
@kyujin-cho Please update this PR.

File: src/ai/backend/manager/registry.py [hunk 1]
```diff
@@ -484,13 +484,13 @@ def __init__(self, config_server, dbpool,
...
                 matched_mounts.add(item['name'])
-                determined_mounts.append((
+                determined_mounts.append({
                     item['name'],
                     item['host'],
                     item['id'].hex,
                     item['permission'].value,
                     item['unmanaged_path'] if item['unmanaged_path'] else ''
-                ))
+                })
```
Comment (achimnol) [inline src/ai/backend/manager/registry.py]: Why is this changed to a list of sets from a list of tuples?

File: src/ai/backend/manager/registry.py [hunk 2]
```diff
@@ -664,6 +665,26 @@ def __init__(self, config_server, dbpool,
...
+                conn, user_uuid,
+                domain_name=sess_ctx.domain_name,
+                allowed_vfolder_types=['user', 'group'],
+                extra_vf_conds=(vfolders.c.name == '.local'))
+            for folder in matched_vfolders:
+                if (folder['group'] is not None and folder['group'] == str(sess_ctx.group_id))\
+                        or package_directory is None:
+                    package_directory = [folder['id'].hex,
+                                         None if folder['group'] is None else user_uuid.hex,
+                                         folder['host']]
```
Comment (achimnol) [inline src/ai/backend/manager/registry.py]: Could you explain how `package_directry` variable is added instead of reusing existing `mount_map`?

File: src/ai/backend/manager/scheduler/dispatcher.py [hunk 3]
```diff
@@ -349,6 +349,7 @@ def __init__(
                 kernels.c.sess_id,
                 kernels.c.access_key,
                 kernels.c.domain_name,
+                kernels.c.user_uuid,
```
Comment (achimnol) [inline src/ai/backend/manager/scheduler/dispatcher.py]: Why is this line added though the column is not referenced below instantiation of `PendingSession`?

File: src/ai/backend/gateway/vfolder.py [hunk 4]
```diff
@@ -569,7 +569,7 @@ def get_folder_hostpath(row: VFolderRow, app):
                 if not entry['is_owner']:
                     raise InvalidAPIParameters(
                         'Cannot change the name of a vfolder '
-                        'that is not owned by me.')
+                        'that is not owned by myself.')
```
Comment (achimnol) [inline src/ai/backend/gateway/vfolder.py]: 👍

File: src/ai/backend/gateway/vfolder.py [hunk 5]
```diff
@@ -1360,7 +1360,7 @@ def _scan():
                 if not entry['is_owner']:
                     raise InvalidAPIParameters(
                         'Cannot delete the vfolder '
-                        'that is not owned by me.')
+                        'that is not owned by myself.')
```
Comment (achimnol) [inline src/ai/backend/gateway/vfolder.py]: :+1:

File: src/ai/backend/manager/registry.py (hunk 1 above)
Comment (kyujin-cho) [reply src/ai/backend/manager/registry.py]: Not an intended type change; looks like a mistake while editing other codes. I'll fix it at next commit.

File: src/ai/backend/manager/scheduler/dispatcher.py (hunk 3 above)
Comment (kyujin-cho) [reply src/ai/backend/manager/scheduler/dispatcher.py]: I've forgot to remove while refactoring whole commit. :(

File: src/ai/backend/manager/registry.py (hunk 2 above)
Comment (kyujin-cho) [reply src/ai/backend/manager/registry.py]: This was essential when agent manipulated per-user package directory itself. But since manager now controls folder creation, it'll be better to reuse mount_map instead of adding additional variable. I'll fix it at next commit.
